webapp/src/js/controllers/inbox.js

```diff
@@ -199,17 +199,18 @@ const LAST_REPLICATED_SEQ_KEY = require('../bootstrapper/purger').LAST_REPLICATE
     });
 
     $rootScope.$on('$stateChangeStart', function (event, toState, toParams, fromState){
-      if (toState.name.indexOf('reports') === -1 || toState.name.indexOf('contacts') === -1 || toState.name.indexOf('tasks') === -1 || toState.name.indexOf('messages.detail') === -1) {
-        $scope.unsetSelected();
-      }
-      if (toState.name.indexOf('tasks.detail') === -1) {
-        Enketo.unload($scope.form);
-        $scope.unsetSelected();
-      }
-      if (toState.name.split('.')[0] !== fromState.name.split('.')[0]){
-        $scope.$broadcast('ClearSelected');
-      }
       if (!$scope.enketoStatus.edited){
+        if (toState.name.indexOf('reports') === -1 || toState.name.indexOf('contacts') === -1 || toState.name.indexOf('tasks') === -1 || toState.name.indexOf('messages.detail') === -1) {
```

Comment:
I resisted the urge to remove these if clauses, even if they're doing exactly nothing. 
- first clause is always true, unless your to state is `reports.contacts.tasks.messages.detail`
- second clause is useless cause only the child scope has a `form` property and `$scope.unsetSelected()` was already called in the first clause
- `$scope.unsetSelected()` also broadcasts `ClearSelected`

Reply:
The third one may be useful because it'll fire at different times. It should be an `else if` though.